pkg/controllers/v1alpha2/mpi_job_controller.go

```diff
@@ -440,12 +439,15 @@ func (c *MPIJobController) syncHandler(key string) error {
 		return err
 	}
 	// We're done if the launcher either succeeded or failed.
-	done := launcher != nil && (launcher.Status.Succeeded == 1 || launcher.Status.Failed == 1)
+	done := launcher != nil && isJobFinished(launcher)
```

Comment:
let's use code [#L24](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/utils.go#L24) from k8s to judge if job has finished.